tests/cache/tests.py

```diff
@@ -1697,6 +1702,76 @@ def test_empty_cache_file_considered_expired(self):
             self.assertIs(cache._is_expired(fh), True)
 
 
+@unittest.skipUnless(RedisCache_params, "Redis backend not configured")
+@override_settings(CACHES=caches_setting_for_tests(
+    base=RedisCache_params,
+    exclude=redis_excluded_caches,
+))
+class RedisCacheTests(BaseCacheTests, TestCase):
+
+    def setUp(self):
+        import redis
+        super().setUp()
+        self.lib = redis
+
+    def test_cache_client_class(self):
+        self.assertIs(cache._class, RedisCacheClient)
+        self.assertIsInstance(cache._cache, RedisCacheClient)
+
+    def test_get_backend_timeout_method(self):
+        positive_timeout = 10
+        positive_backend_timeout = cache.get_backend_timeout(positive_timeout)
+        self.assertEqual(positive_backend_timeout, positive_timeout)
+
+        negative_timeout = -5
+        negative_backend_timeout = cache.get_backend_timeout(negative_timeout)
+        self.assertEqual(negative_backend_timeout, 0)
+
+        none_timeout = None
+        none_backend_timeout = cache.get_backend_timeout(none_timeout)
+        self.assertIsNone(none_backend_timeout)
+
+    def test__get_connection_pool_index(self):
+        pool_index = cache._cache._get_connection_pool_index(write=True)
+        self.assertEqual(pool_index, 0)
+
+        pool_index = cache._cache._get_connection_pool_index(write=False)
+
+        if len(cache._cache._servers) == 1:
+            self.assertEqual(pool_index, 0)
+        else:
+            self.assertGreater(pool_index, 0)
+            self.assertLess(pool_index, len(cache._cache._servers))
+
+    def test_get_connection_pool(self):
+        pool = cache._cache._get_connection_pool(write=True)
+        self.assertIsInstance(pool, self.lib.ConnectionPool)
+
+        pool = cache._cache._get_connection_pool(write=False)
+        self.assertIsInstance(pool, self.lib.ConnectionPool)
+
+    def test_get_client(self):
+        self.assertIsInstance(cache._cache.get_client(), self.lib.Redis)
+
+
+class RedisCachePickleSerializerTests(TestCase):
+
+    def test_serializer_protocol(self):
+        # If protocol is not provided, it defaults to pickle.HIGHEST_PROTOCOL
+        serializer = PickleSerializer()
+        self.assertEqual(serializer._protocol, pickle.HIGHEST_PROTOCOL)
+
+        serializer = PickleSerializer(protocol=3)
+        self.assertEqual(serializer._protocol, 3)
+
+    def test_serializer_loads_dumps(self):
+        serializer = PickleSerializer()
+        test_data = 'foobar'
+
+        dump = serializer.dumps(test_data)
+        self.assertEqual(serializer.loads(dump), test_data)
```

Comment:
We could move these tests to a separate PR, together with `PickleSerializer`, see https://github.com/django/django/pull/14437#discussion_r696343672.

Reply:
I don't think that `BaseSerializer` is necessary. I only want to avoid defining the same class in different places so moving `PickleSerializer` (as described in https://github.com/django/django/pull/14437#discussion_r696343672) is enough.

Reply:
Alright! Updating it